src/main/kotlin/com/app/ui/screens/games/SudokuScreen.kt

```diff
@@ -16,185 +14,250 @@ import androidx.compose.runtime.rememberCoroutineScope
-
-        Row(
-            modifier = Modifier
-                .padding(4.dp),
-            horizontalArrangement = Arrangement.SpaceEvenly
-        ) {
-            Button(
-                onClick = onBack,
-                colors = ButtonDefaults.buttonColors(
-                    backgroundColor = Color(0xFFC5705D),
-                    contentColor = Color.White
+                    },
+                    isEditingNotes,
                 )
-            ) {
-                Text("Main Menu")
             }
-
-            Spacer(modifier = Modifier.width(8.dp))
-
-            // Saving current state of the board to the database
-            Button(
-                onClick = {
-                    scope.launch {
-                        val temp = board
+        }
+        Menu(
+            onBack,
+            isPaused,
+            onClickSaveGame = {
+                scope.launch {
+                    isPaused = true
+                    val temp = board
+                    board = null
+                    board = saveSudoku(temp!!, currentBoard, currentTime)
+                }
+            },
+            onClickNewGame = {
+                scope.launch {
+                    try {
                         board = null
-                        board = saveSudoku(temp)
+                        isPaused = true
+                        currentTime = 0
+                        board = newGame(currentBoard)
+                    } catch (e: Exception) {
+                        println("Error during creating new game: ${e.message}")
                     }
-                },
-                colors = ButtonDefaults.buttonColors(
-                    backgroundColor = Color(0xFFC5705D),
-                    contentColor = Color.White
-                )
-            ) {
-                Text("Save progress")
-            }
-
-            Spacer(modifier = Modifier.width(8.dp))
-
-            // Displaying the basic board to the user.
-            Button(
-                onClick = {
-                    scope.launch {
-                        try {
-                            board = null
-                            board = newGame()
-                        } catch (e: Exception) {
-                            println("Error during creating new game: ${e.message}")
-                        }
-                    }
-                },
-                colors = ButtonDefaults.buttonColors(
-                    backgroundColor = Color(0xFFC5705D),
-                    contentColor = Color.White
-                )
-            ) {
-                Text("New game")
-            }
-        }
+                }
+            },
+            onClickPause = {
+                isPaused = !isPaused
+            },
+            onClickNextGame = {
+                scope.launch {
+                    board = null
+                    isPaused = true
+                    currentTime = 0
+                    currentBoard = (currentBoard + 2) % 4
```

Comment:
to rozwiązanie - o ile dobrze rozumiem -  z "wzorcami" plansz niezbyt mi się podoba, bo wygląda magicznie (czyt. niezrozumiale)